korman/exporter/etlight.py

```diff
@@ -384,25 +390,28 @@ def _prep_for_lightmap(self, bo, toggle):
             uv_textures.active = uvtex
 
             # if the artist hid any UVs, they will not be baked to... fix this now
-            bpy.ops.object.mode_set(mode="EDIT")
-            bpy.ops.uv.reveal()
-            bpy.ops.object.mode_set(mode="OBJECT")
+            with self._set_mode("EDIT"):
+                bpy.ops.uv.reveal()
             self._associate_image_with_uvtex(uv_textures.active, im)
-            bpy.ops.object.mode_set(mode="EDIT")
 
-            # prep the uvtex for lightmapping
-            bpy.ops.mesh.select_all(action="SELECT")
-            bpy.ops.uv.average_islands_scale()
-            bpy.ops.uv.pack_islands()
+            # Meshes with modifiers need to have islands packed to prevent generated vertices
+            # from sharing UVs. Sigh.
+            if self._mesh.is_collapsed(bo):
+                # Danger: uv_base.name -> UnicodeDecodeError (wtf? another blender bug?)
+                self._report.warn("'{}': packing islands in UV Texture '{}' due to modifier collapse",
+                                  bo.name, modifier.uv_map, indent=2)
+                with self._set_mode("EDIT"):
+                    bpy.ops.mesh.select_all(action="SELECT")
+                    bpy.ops.uv.pack_islands(margin=0.01)
         else:
             # same thread, see Sirius's suggestion RE smart unwrap. this seems to yield good
             # results in my tests. it will be good enough for quick exports.
             uvtex = uv_textures.new(self.lightmap_uvtex_name)
+            uv_textures.active = uvtex
             self._associate_image_with_uvtex(uvtex, im)
-            bpy.ops.object.mode_set(mode="EDIT")
-            bpy.ops.mesh.select_all(action="SELECT")
-            bpy.ops.uv.smart_project()
-        bpy.ops.object.mode_set(mode="OBJECT")
+            with self._set_mode("EDIT"):
+                bpy.ops.mesh.select_all(action="SELECT")
```

Comment:
```suggestion
                bpy.ops.mesh.select_all(action="SELECT")
                bpy.ops.uv.select_all(action="SELECT")
```
Otherwise island packing won't work.

Reply:
Will this change also need to be replicated between lines 404 and 405?

Reply:
Nice catch - I actually clicked the wrong line when reviewing the second time. Yes, `bpy.ops.uv.select_all(action="SELECT")` must be placed before `bpy.ops.uv.pack_islands`, NOT before `bpy.ops.uv.smart_project` (the former operates on selected UV vertices, the latter on selected mesh vertices). Sorry for the confusion.